psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,148 @@
+        }
+        else if (SERVICE_PROVIDER.equals(creditCardNumbers.get(0))) {
+            adDetailsList = new ArrayList<>(creditCardNumbers.size());
+            for (int i = 1; i < creditCardNumbers.size(); i++) {
+                AdDetails adDetails = new AdDetails(creditCardNumbers.get(i));
                 adDetailsList.add(adDetails);
             }
             inputData.setAdDetailsList(adDetailsList);
         }
+        return inputData;
+    }
 
-        //retrieve the attestation object from host immediately after connecting to host
-        byte[] attestationBytes = new byte[fromHost.readInt()];
-        fromHost.readFully(attestationBytes);
-
-        //convert byte[] received from host to EnclaveInstanceInfo object
-        EnclaveInstanceInfo instanceInfo = EnclaveInstanceInfo.deserialize(attestationBytes);
-
-        //verify attestation received by enclave against the enclave code hash which we have
-        EnclaveConstraint.parse("S:"+ constraint +" PROD:1 SEC:INSECURE" ).check(instanceInfo);
-
-        //create a dummy key pair for sending via mail to enclave
-        PrivateKey key = Curve25519PrivateKey.random();
-
-        //create PostOffice specifying - clients public key, topic name , enclaves public key
-        PostOffice postOffice = instanceInfo.createPostOffice(key, UUID.randomUUID().toString());
-
-        //encrypt the message using enclave's public key
-        byte[] encryptedRequest = postOffice.encryptMail(serializeMessage(inputData).getBuffer());
-
-        //send the encrypted mail to host to relay it to enclave
-        toHost.writeInt(encryptedRequest.length);
-        toHost.write(encryptedRequest);
-
-        //get the reply back from host via the socket
-        byte[] encryptedReply = new byte[fromHost.readInt()];
-        fromHost.readFully(encryptedReply);
-
-        //use Post Office to decrypt back the mail sent by the enclave
-        EnclaveMail mail = postOffice.decryptMail(encryptedReply);
-
-        System.out.println("Ad Conversion Rate : " + new String(mail.getBodyAsBytes()) +"%");
-
-        toHost.close();
-        fromHost.close();
+    private static class EnclaveConstraintConverter implements CommandLine.ITypeConverter<EnclaveConstraint> {
+        @Override
+        public EnclaveConstraint convert(String value) {
+            return EnclaveConstraint.parse(value);
+        }
     }
 
-    /**
-     * Use Kryo to serialize inputs from client to enclave
-     */
-    private static Output serializeMessage(InputData listOfCreditCardNumbers){
-        Kryo kryo = new Kryo();
-        Output output = new Output(new ByteArrayOutputStream());
-        kryo.register(InputData.class, new InputDataSerializer());
-        kryo.writeObject(output, listOfCreditCardNumbers);
-        output.close();
-        return output;
+    public static void main(String... args) {
+        int exitCode = new CommandLine(new Client()).execute(args);
```

Comment:
You should call `System.exit(exitCode);`.

Reply:
done